In [2]:
import pandas as pd

Read Data

In [3]:
df = pd.read_csv("data\data.csv", encoding='latin-1')
df

Tweet  HS  Abusive  \
0      - disaat semua cowok berusaha melacak perhatia...   1        1   
1      RT USER: USER siapa yang telat ngasih tau elu?...   0        1   
2      41. Kadang aku berfikir, kenapa aku tetap perc...   0        0   
3      USER USER AKU ITU AKU\n\nKU TAU MATAMU SIPIT T...   0        0   
4      USER USER Kaum cebong kapir udah keliatan dong...   1        1   
...                                                  ...  ..      ...   
13164  USER jangan asal ngomong ndasmu. congor lu yg ...   1        1   
13165                       USER Kasur mana enak kunyuk'   0        1   
13166  USER Hati hati bisu :( .g\n\nlagi bosan huft \...   0        0   
13167  USER USER USER USER Bom yang real mudah terdet...   0        0   
13168  USER Mana situ ngasih(": itu cuma foto ya kuti...   1        1   

       HS_Individual  HS_Group  HS_Religion  HS_Race  HS_Physical  HS_Gender  \
0                  1         0            0        0            0          0   
1                  0         0            0        0            0          0   
2                  0         0            0        0            0          0   
3                  0         0            0        0            0          0   
4                  0         1            1        0            0          0   
...              ...       ...          ...      ...          ...        ...   
13164              1         0            0        0            1          0   
13165              0         0            0        0            0          0   
13166              0         0            0        0            0          0   
13167              0         0            0        0            0          0   
13168              1         0            0        0            0          0   

       HS_Other  HS_Weak  HS_Moderate  HS_Strong  
0             1        1            0          0  
1             0        0            0          0  
2             0        0            0          0  
3             0        0            0          0  
4             0        0            1          0  
...         ...      ...          ...        ...  
13164         0        1            0          0  
13165         0        0            0          0  
13166         0        0            0          0  
13167         0        0            0          0  
13168         1        1            0          0  

[13169 rows x 13 columns]

In [4]:
df.head()

Tweet  HS  Abusive  \
0  - disaat semua cowok berusaha melacak perhatia...   1        1   
1  RT USER: USER siapa yang telat ngasih tau elu?...   0        1   
2  41. Kadang aku berfikir, kenapa aku tetap perc...   0        0   
3  USER USER AKU ITU AKU\n\nKU TAU MATAMU SIPIT T...   0        0   
4  USER USER Kaum cebong kapir udah keliatan dong...   1        1   

   HS_Individual  HS_Group  HS_Religion  HS_Race  HS_Physical  HS_Gender  \
0              1         0            0        0            0          0   
1              0         0            0        0            0          0   
2              0         0            0        0            0          0   
3              0         0            0        0            0          0   
4              0         1            1        0            0          0   

   HS_Other  HS_Weak  HS_Moderate  HS_Strong  
0         1        1            0          0  
1         0        0            0          0  
2         0        0            0          0  
3         0        0            0          0  
4         0        0            1          0

In [5]:
df.tail()

Tweet  HS  Abusive  \
13164  USER jangan asal ngomong ndasmu. congor lu yg ...   1        1   
13165                       USER Kasur mana enak kunyuk'   0        1   
13166  USER Hati hati bisu :( .g\n\nlagi bosan huft \...   0        0   
13167  USER USER USER USER Bom yang real mudah terdet...   0        0   
13168  USER Mana situ ngasih(": itu cuma foto ya kuti...   1        1   

       HS_Individual  HS_Group  HS_Religion  HS_Race  HS_Physical  HS_Gender  \
13164              1         0            0        0            1          0   
13165              0         0            0        0            0          0   
13166              0         0            0        0            0          0   
13167              0         0            0        0            0          0   
13168              1         0            0        0            0          0   

       HS_Other  HS_Weak  HS_Moderate  HS_Strong  
13164         0        1            0          0  
13165         0        0            0          0  
13166         0        0            0          0  
13167         0        0            0          0  
13168         1        1            0          0

In [6]:
df.shape

(13169, 13)

Memeriksa dan menghilangkan duplikat

In [7]:
df.duplicated().sum()

125

In [8]:
df = df.drop_duplicates()

In [9]:
df.isna()

Tweet     HS  Abusive  HS_Individual  HS_Group  HS_Religion  HS_Race  \
0      False  False    False          False     False        False    False   
1      False  False    False          False     False        False    False   
2      False  False    False          False     False        False    False   
3      False  False    False          False     False        False    False   
4      False  False    False          False     False        False    False   
...      ...    ...      ...            ...       ...          ...      ...   
13164  False  False    False          False     False        False    False   
13165  False  False    False          False     False        False    False   
13166  False  False    False          False     False        False    False   
13167  False  False    False          False     False        False    False   
13168  False  False    False          False     False        False    False   

       HS_Physical  HS_Gender  HS_Other  HS_Weak  HS_Moderate  HS_Strong  
0            False      False     False    False        False      False  
1            False      False     False    False        False      False  
2            False      False     False    False        False      False  
3            False      False     False    False        False      False  
4            False      False     False    False        False      False  
...            ...        ...       ...      ...          ...        ...  
13164        False      False     False    False        False      False  
13165        False      False     False    False        False      False  
13166        False      False     False    False        False      False  
13167        False      False     False    False        False      False  
13168        False      False     False    False        False      False  

[13044 rows x 13 columns]

Menampilkan info tentang data

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 13044 entries, 0 to 13168
Data columns (total 13 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Tweet          13044 non-null  object
 1   HS             13044 non-null  int64 
 2   Abusive        13044 non-null  int64 
 3   HS_Individual  13044 non-null  int64 
 4   HS_Group       13044 non-null  int64 
 5   HS_Religion    13044 non-null  int64 
 6   HS_Race        13044 non-null  int64 
 7   HS_Physical    13044 non-null  int64 
 8   HS_Gender      13044 non-null  int64 
 9   HS_Other       13044 non-null  int64 
 10  HS_Weak        13044 non-null  int64 
 11  HS_Moderate    13044 non-null  int64 
 12  HS_Strong      13044 non-null  int64 
dtypes: int64(12), object(1)
memory usage: 1.4+ MB


In [11]:
df.describe()

HS       Abusive  HS_Individual      HS_Group   HS_Religion  \
count  13044.000000  13044.000000   13044.000000  13044.000000  13044.000000   
mean       0.423030      0.383701       0.271389      0.151641      0.060488   
std        0.494059      0.486305       0.444693      0.358686      0.238397   
min        0.000000      0.000000       0.000000      0.000000      0.000000   
25%        0.000000      0.000000       0.000000      0.000000      0.000000   
50%        0.000000      0.000000       0.000000      0.000000      0.000000   
75%        1.000000      1.000000       1.000000      0.000000      0.000000   
max        1.000000      1.000000       1.000000      1.000000      1.000000   

            HS_Race   HS_Physical     HS_Gender      HS_Other       HS_Weak  \
count  13044.000000  13044.000000  13044.000000  13044.000000  13044.000000   
mean       0.043162      0.024686      0.023306      0.284115      0.256670   
std        0.203229      0.155171      0.150879      0.451009      0.436812   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        0.000000      0.000000      0.000000      0.000000      0.000000   
50%        0.000000      0.000000      0.000000      0.000000      0.000000   
75%        0.000000      0.000000      0.000000      1.000000      1.000000   
max        1.000000      1.000000      1.000000      1.000000      1.000000   

        HS_Moderate     HS_Strong  
count  13044.000000  13044.000000  
mean       0.130175      0.036185  
std        0.336509      0.186758  
min        0.000000      0.000000  
25%        0.000000      0.000000  
50%        0.000000      0.000000  
75%        0.000000      0.000000  
max        1.000000      1.000000

Menghitung total karakter dan total kata dari kolom Tweet

In [12]:
df['total_char']= df.Tweet.apply(len)

C:\Users\yazid\AppData\Local\Temp\ipykernel_7336\2817185475.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['total_char']= df.Tweet.apply(len)


In [13]:
df.head()

Tweet  HS  Abusive  \
0  - disaat semua cowok berusaha melacak perhatia...   1        1   
1  RT USER: USER siapa yang telat ngasih tau elu?...   0        1   
2  41. Kadang aku berfikir, kenapa aku tetap perc...   0        0   
3  USER USER AKU ITU AKU\n\nKU TAU MATAMU SIPIT T...   0        0   
4  USER USER Kaum cebong kapir udah keliatan dong...   1        1   

   HS_Individual  HS_Group  HS_Religion  HS_Race  HS_Physical  HS_Gender  \
0              1         0            0        0            0          0   
1              0         0            0        0            0          0   
2              0         0            0        0            0          0   
3              0         0            0        0            0          0   
4              0         1            1        0            0          0   

   HS_Other  HS_Weak  HS_Moderate  HS_Strong  total_char  
0         1        1            0          0         138  
1         0        0            0          0         120  
2         0        0            0          0         254  
3         0        0            0          0          75  
4         0        0            1          0          89

In [14]:
df['total_word']= df.Tweet.apply(lambda sent: len(sent.split()))

C:\Users\yazid\AppData\Local\Temp\ipykernel_7336\3117698956.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['total_word']= df.Tweet.apply(lambda sent: len(sent.split()))


In [15]:
df.head()

Tweet  HS  Abusive  \
0  - disaat semua cowok berusaha melacak perhatia...   1        1   
1  RT USER: USER siapa yang telat ngasih tau elu?...   0        1   
2  41. Kadang aku berfikir, kenapa aku tetap perc...   0        0   
3  USER USER AKU ITU AKU\n\nKU TAU MATAMU SIPIT T...   0        0   
4  USER USER Kaum cebong kapir udah keliatan dong...   1        1   

   HS_Individual  HS_Group  HS_Religion  HS_Race  HS_Physical  HS_Gender  \
0              1         0            0        0            0          0   
1              0         0            0        0            0          0   
2              0         0            0        0            0          0   
3              0         0            0        0            0          0   
4              0         1            1        0            0          0   

   HS_Other  HS_Weak  HS_Moderate  HS_Strong  total_char  total_word  
0         1        1            0          0         138          25  
1         0        0            0          0         120          21  
2         0        0            0          0         254          37  
3         0        0            0          0          75          14  
4         0        0            1          0          89          14

Menggunakan univariate analysis:
    1.  Mean (rata-rata)
    2.  Median (nilai tengah)
    3.  Mode (yang paling sering muncul)

In [16]:
df.total_char.mean()

114.12434835939895

In [17]:
df.total_char.median()

100.0

In [18]:
df.total_char.mode()

0    254
Name: total_char, dtype: int64

Menghitung penyebaran data

In [19]:
range_total_char = df.total_char.max() - df.total_char.min()
range_total_char

557

In [20]:
#quartile dan interquartile range

p0 = df.total_char.min() #nilai minimum
p100 = df.total_char.max() #nilai maksimum

#cari quartile 1
q1 = df.total_char.quantile(0.25)

#cari quartile 2
q2 = df.total_char.quantile(0.5)

#cari quartile 3
q3 = df.total_char.quantile(0.75)

#cari IQR
iqr = q3 - q1

#batas bawah
lower_limit = q1 - 1.5 * iqr

#batas atas
upper_limit = q3 + 1.5 * iqr

In [21]:
print("batas bawah 'total_char' : ", lower_limit)
print("nilai minimum: ",p0)

if lower_limit<p0:
  print("tidak ada outlier dari sisi bawah")
else:
  print("ada outlier dari sisi bawah")

print("batas atas 'total_char' : ", upper_limit)
print("nilai maksimum: ",p100)

if upper_limit>p100:
  print("tidak ada outlier dari sisi atas")
else:
  print("ada outlier dari sisi atas")

batas bawah 'total_char' :  -80.5
nilai minimum:  4
tidak ada outlier dari sisi bawah
batas atas 'total_char' :  291.5
nilai maksimum:  561
ada outlier dari sisi atas


In [31]:
range_total_word = df.total_word.max() - df.total_word.min()
range_total_word

51

In [32]:
#quartile dan interquartile range

p0w = df.total_word.min() #nilai minimum
p100w = df.total_word.max() #nilai maksimum

#cari quartile 1
q1w = df.total_word.quantile(0.25)

#cari quartile 2
q2w = df.total_word.quantile(0.5)

#cari quartile 3
q3w = df.total_word.quantile(0.75)

#cari IQR
iqrw = q3 - q1

#batas bawah
lower_limit_word = q1w - 1.5 * iqrw

#batas atas
upper_limit_word = q3w + 1.5 * iqrw

In [33]:
print("batas bawah 'total_word' : ", lower_limit_word)
print("nilai minimum: ",p0w)

if lower_limit_word<p0w:
  print("tidak ada outlier dari sisi bawah")
else:
  print("ada outlier dari sisi bawah")

print("batas atas 'total_word' : ", upper_limit_word)
print("nilai maksimum: ",p100w)

if upper_limit_word>p100w:
  print("tidak ada outlier dari sisi atas")
else:
  print("ada outlier dari sisi atas")

batas bawah 'total_word' :  -130.5
nilai minimum:  1
tidak ada outlier dari sisi bawah
batas atas 'total_word' :  162.5
nilai maksimum:  52
tidak ada outlier dari sisi atas


In [22]:
#variance
print(df.total_char.mean())
print(df.total_char.var())

114.12434835939895
4775.035061485935


In [23]:
#standar deviasi
print(df.total_char.mean())
print(df.total_char.std())

114.12434835939895
69.10162850096903


Mendeskripsikan persebaran data melalui grafik

In [24]:
df.total_char.skew()

0.6936467482723248

In [25]:
df.total_char.kurtosis()

-0.4022908274147734

In [34]:
df.total_word.skew()

0.7898198026333267

In [35]:
df.total_word.kurtosis()

-0.15170747414709895

Melihat korelasi antar kolom

In [28]:
df.corr()

C:\Users\yazid\AppData\Local\Temp\ipykernel_7336\1134722465.py:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  df.corr()


HS   Abusive  HS_Individual  HS_Group  HS_Religion  \
HS             1.000000  0.365292       0.712754  0.493752     0.296328   
Abusive        0.365292  1.000000       0.286353  0.148142     0.069610   
HS_Individual  0.712754  0.286353       1.000000 -0.258028     0.033899   
HS_Group       0.493752  0.148142      -0.258028  1.000000     0.366139   
HS_Religion    0.296328  0.069610       0.033899  0.366139     1.000000   
HS_Race        0.248039 -0.028722      -0.000672  0.342486     0.094862   
HS_Physical    0.185798  0.196547       0.214011 -0.009406    -0.003061   
HS_Gender      0.180403  0.187413       0.183402  0.021111     0.024751   
HS_Other       0.735728  0.261476       0.636195  0.224657    -0.159848   
HS_Weak        0.686258  0.310169       0.962826 -0.248436     0.033490   
HS_Moderate    0.451792  0.188095      -0.236100  0.915018     0.345290   
HS_Strong      0.226287 -0.098015       0.058995  0.238550     0.083432   
total_char    -0.148112 -0.211936      -0.140442 -0.029893    -0.004901   
total_word    -0.128527 -0.193352      -0.119116 -0.029356     0.004380   

                HS_Race  HS_Physical  HS_Gender  HS_Other   HS_Weak  \
HS             0.248039     0.185798   0.180403  0.735728  0.686258   
Abusive       -0.028722     0.196547   0.187413  0.261476  0.310169   
HS_Individual -0.000672     0.214011   0.183402  0.636195  0.962826   
HS_Group       0.342486    -0.009406   0.021111  0.224657 -0.248436   
HS_Religion    0.094862    -0.003061   0.024751 -0.159848  0.033490   
HS_Race        1.000000    -0.028927  -0.020306 -0.133800 -0.007345   
HS_Physical   -0.028927     1.000000   0.054019 -0.100225  0.220971   
HS_Gender     -0.020306     0.054019   1.000000 -0.097315  0.190753   
HS_Other      -0.133800    -0.100225  -0.097315  1.000000  0.605078   
HS_Weak       -0.007345     0.220971   0.190753  0.605078  1.000000   
HS_Moderate    0.238469     0.000123   0.027826  0.225597 -0.227323   
HS_Strong      0.243672    -0.025535  -0.019047  0.124610 -0.113858   
total_char    -0.056811    -0.043631  -0.025737 -0.109286 -0.131338   
total_word    -0.048205    -0.035267  -0.018621 -0.101147 -0.108383   

               HS_Moderate  HS_Strong  total_char  total_word  
HS                0.451792   0.226287   -0.148112   -0.128527  
Abusive           0.188095  -0.098015   -0.211936   -0.193352  
HS_Individual    -0.236100   0.058995   -0.140442   -0.119116  
HS_Group          0.915018   0.238550   -0.029893   -0.029356  
HS_Religion       0.345290   0.083432   -0.004901    0.004380  
HS_Race           0.238469   0.243672   -0.056811   -0.048205  
HS_Physical       0.000123  -0.025535   -0.043631   -0.035267  
HS_Gender         0.027826  -0.019047   -0.025737   -0.018621  
HS_Other          0.225597   0.124610   -0.109286   -0.101147  
HS_Weak          -0.227323  -0.113858   -0.131338   -0.108383  
HS_Moderate       1.000000  -0.074958   -0.004277   -0.006556  
HS_Strong        -0.074958   1.000000   -0.076927   -0.074699  
total_char       -0.004277  -0.076927    1.000000    0.955358  
total_word       -0.006556  -0.074699    0.955358    1.000000

Conclusion

1. Dalam statistik deskriptif untuk total karakter menunjukkan adanya outlier di bagian atas tetapi tidak begitu banyak.
2. Dalam statistik deskriptif untuk total kata menunjukkan tidak ada outlier di sisi atas maupun bawah.
3. Visualisasi data menunjukkan:
    - 13169 baris data dengan 13 kolom
    - Total karakter memiliki panjang 4 sampai 561 karakter
    - Total kata memiliki panjang 1 sampai 52 kata
4. Dalam statistik deskriptif terlihat bahwa variabel total_char dan total_word memiliki korelasi positif. Semakin banyak total_char maka semakin banyak total_word.